tests/test_executor.py

```diff
@@ -0,0 +1,99 @@
+########
+# Copyright (c) 2017 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import time
+
+import pytest
+
+from aria.orchestrator import workflow_runner
+
+from aria_plugin import executor
+from aria_plugin.exceptions import AriaWorkflowError
+
+
+@pytest.fixture
+def mocked_env(mocker):
+    mocked_env = mocker.MagicMock()
+    mocked_env.model_storage = 'model_storage'
+    mocked_env.resource_storage = 'resource_storage'
+    mocked_env.plugin_manager = 'plugin_manager'
+    mocked_env.service.id = 'service_id'
+    mocked_env.ctx_logger = mocker.MagicMock()
+
+    return mocked_env
+
+
+def _patch_runner(mocker, success=True):
+    mock_runner = mocker.MagicMock()
+
+    mock_execution = mocker.MagicMock()
+    mock_execution.SUCCEEDED = 'pass'
+    mock_execution.status = 'pass' if success else 'fail'
+    mock_runner.execution = mock_execution
+
+    mocker.patch('aria.orchestrator.workflow_runner.WorkflowRunner',
+                 return_value=mock_runner)
+
+    return mock_runner
+
+
+def test_successful_execute(mocked_env, mocker):
+    mocker.patch('aria.cli.logger.ModelLogIterator', return_value=[])
+    mock_runner = _patch_runner(mocker)
+
+    executor.execute(mocked_env, 'workflow_name')
+
+    workflow_runner.WorkflowRunner.assert_called_once_with(
+        'model_storage', 'resource_storage', 'plugin_manager',
+        service_id='service_id',
+        workflow_name='workflow_name',
+        executor=mocker.ANY
+    )
+
+    mock_runner.execute.assert_called_once_with()
```

Comment:
no need for `with`

Reply:
`assert_called_once_with()` is a stronger assertion. Removing it wouldn't be quite the same test.